openml/extensions/sklearn/extension.py

```diff
@@ -1268,154 +1249,50 @@ def _prediction_to_probabilities(
             user_defined_measures['wall_clock_time_millis'] = (modelfit_dur_walltime
                                                                + modelpredict_duration_walltime)
 
-        # add client-side calculated metrics. These is used on the server as
-        # consistency check, only useful for supervised tasks
-        def _calculate_local_measure(sklearn_fn, openml_name):
-            user_defined_measures[openml_name] = sklearn_fn(test_y, pred_y)
-
-        # Task type specific outputs
-        arff_datacontent = []
-
         if isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
 
+            if classes is None:
+                raise TypeError("Argument classes must not be of type 'None'")
+
             try:
-                proba_y = model_copy.predict_proba(test_x)
+                proba_y = model_copy.predict_proba(X_test)
             except AttributeError:
-                proba_y = _prediction_to_probabilities(pred_y, list(model_classes))
+                proba_y = _prediction_to_probabilities(pred_y, list(classes))
+
+            if proba_y.shape[1] != len(classes):
+                # Remap the probabilities in case there was a class missing at training time
+                # By default, the classification targets are mapped to be zero-based indices to the
+                # actual classes. Therefore, the model_classes contain the correct indices to the
+                # correct probability array (the actualy array might be incorrect if there are some
+                # classes not present during train time).
+                proba_y_new = np.zeros((proba_y.shape[0], len(classes)))
+                for idx, model_class in enumerate(model_classes):
```

Comment:
I don't understand what you're trying to do here.
If targets are mapped to be zero-based indices, wouldn't `enumerate(model_classes)` just evaluate to `(1,1), (2,2), ..., (Kt, Kt)` where `Kt` is the number of classes in the training data?
If so why not do either
```
proba_y_new = np.zeros((proba_y.shape[0], len(classes)))
proba_y_new[:proba_y.shape[0], :proba_y.shape[1]] = proba_y
```
or use `np.hstack` to pad zero-columns?

Reply:
The zero-columns might need to be in the middle of the array. I added an example to the comment.